### PR TITLE
Fix day change BTC

### PR DESCRIPTION
**Pull request by maromalo**
It now uses reports instead of the database, as the latter does not store old coin_value (at least on tntwist's branch).

### CODE REVIEW

**Comment by titulebolide** [comment]
Ho this looks promising!
I'll try it out and review it asap

File: btb_manager_telegram/buttons.py [hunk 1]
```diff
@@ -101,36 +83,74 @@ def current_value():
...
 
+                reports = get_previous_reports()
+                reports.reverse()
+
+                query_1_day, query_7_day = [], []
+                try:
+                    query_1_day_done = False
+                    for r in reports:
+                        if (
+                            r["time"] < int(last_update.timestamp()) - 86400
```
Comment (DmytroLitvinov) [inline btb_manager_telegram/buttons.py:94]: `86400` what does that magic number mean?

File: btb_manager_telegram/buttons.py [hunk 2]
```diff
@@ -101,36 +83,74 @@ def current_value():
...
+                            and not query_1_day_done
+                        ):
+                            query_1_day = (
+                                r["balances"][current_coin],
+                                r["tickers"][current_coin],
+                                r["tickers"]["BTC"],
+                            )
+                            query_1_day_done = True
+                        if r["time"] < int(last_update.timestamp()) - 604800:
```
Comment (DmytroLitvinov) [inline btb_manager_telegram/buttons.py:103]: `604800` what does that magic number mean?

File: btb_manager_telegram/buttons.py (hunk 1 above)
Comment (titulebolide) [reply btb_manager_telegram/buttons.py:94]: it is the number of seconds in one day

File: btb_manager_telegram/buttons.py (hunk 2 above)
Comment (titulebolide) [reply btb_manager_telegram/buttons.py:103]: it is the number of seconds in one week

Comment (titulebolide) [reply btb_manager_telegram/buttons.py:103]: i'm not a fan of using `60*60*24*7`. But we could be using datetime.timedelta(days=7).total_seconds()

**Comment by titulebolide** [comment]
Well
image
It is promising though, i'll have do dive into what's happening.
Besides that we'd have to handle the case where no report were available to compare with (e.g. if the reports are available up to 3 days ago but no older)

File: btb_manager_telegram/buttons.py (hunk 2 above)
Comment (maromalo) [reply btb_manager_telegram/buttons.py:103]: I should have probably left a comment, but yeah, using timedelta would be more readable. I'll be editing it in just a sec

**Comment by maromalo** [comment]
Damn, it did work on my end, probably cause I've been stuck on a coin for about a month
I agree, when there's no report to compare with it should raise a KeyError (in `r["balances"][current_coin]`) and pass, leaving the query empty, but there may be a small amount of balance of that coin left from a previous trade, in which case that won't work, and I'm not sure how to handle that yet

**Comment by titulebolide** [comment]
I'm on it. We can use total_udst from the reports

**Comment by titulebolide** [comment]
I let you check my last commit (i didnt ran black lol)
I used total_usdt instead, and removed the try/except statements that makes some things fail silently which is very dangerous

**Comment by titulebolide** [comment]
If you're not happy with it i'll revert it

**Comment by titulebolide** [comment]
ok nice! i'll clean a bit and add some exceptions handling and we'll be good to go!
@all-contributors please add @maromalo for bugs, code

**Comment by titulebolide** [comment]
ok nice! i'll clean a bit and add some exceptions handling and we'll be good to go!
@all-contributors please add @maromalo for bug, code

**Comment by titulebolide** [comment]
ok nice! i'll clean a bit and add some exceptions handling and we'll be good to go!

**Comment by titulebolide** [comment]
@all-contributors please add @maromalo for bug, code

**Comment by maromalo** [comment]
You forgot to multiply by 100 to show it as a percentile (for a moment I feared I had only moved 0.01% in the last week lol)

**Comment by titulebolide** [comment]
Oh boy yeah thanks :)

**Comment by titulebolide** [comment]
So i've changed the logic of the retreival of the right report in order for it to be more condensed. I make it go to test phase

**Pull request merged**